src/ConstructStyleSheet.js

```diff
@@ -135,7 +119,43 @@ export default class ConstructStyleSheet {
   }
 }
 
+// Implement all methods from the base CSSStyleSheet constructor as
+// a proxy to the raw style element created during construction.
+cssStyleSheetMethods.forEach(method => {
+  ConstructStyleSheet.prototype[method] = function() {
+    if (!sheetMetadataRegistry.has(this)) {
+      throw new Error(
+        `Failed to execute '${method}' on 'CSSStyleSheet': Can't call ${method} on non-constructed CSSStyleSheets.`,
+      )
+    }
+
+    const args = arguments;
+    const { adopters, actions, basicStyleElement } = sheetMetadataRegistry.get(this);
+    const result = basicStyleElement.sheet[method].apply(basicStyleElement.sheet, args);
+
+    adopters.forEach(styleElement => {
+      if (styleElement.sheet) {
+        styleElement.sheet[method].apply(styleElement.sheet, args);
+      }
+    });
+
+    actions.push([method, args]);
+
+    return result;
```

Comment:
Note: this is pretty much an exact copy of the code I removed above, and this is the most significant part of the changes. Here are some notes about why I made this change:

- During init, we set [`window.CSSStyleSheet = ConstructStyleSheet`](https://github.com/calebdwilliams/construct-style-sheets/blob/master/src/index.js#L8), so I feel like the class itself should implement the full API.
- In Chrome, non-constructed stylesheets cannot be adopted. I'm not sure if this is to spec or not:
    ```js
    const style = document.createElement('style');
    document.head.appendChild(style);
    document.adoptedStyleSheets = [ style.sheet ]; // error - failed to convert value to CSSStyleSheet
    ```

Reply:
> In Chrome, non-constructed stylesheets cannot be adopted. I'm not sure if this is to spec or not:

Yeah, that's by design. Does your PR change that behavior?

Reply:
The previous code extended the default CSSStyleSheet (referred to as OldCSSStyleSheet throughout the code) to include `replace`, `replaceSync`, and apply all changes (`addRule` and friends) to all adopted stylesheets. I feel like that behavior is wrong as it makes non-constructed stylesheets behave like constructed ones. So yes, my PR does change that.

Reply:
OK, just to make sure I'm not being overly-vague, I want to outline my current beliefs:

- In Chrome non-constructed stylesheets will throw if they call `replace` or `replaceSync` although oddly enough they will still perform the replacement. Not sure if that's intentional or not, maybe @rakina can help?
- Non-constructed stylesheets should not be adoptable per the spec, trying to do so should throw with `Uncaught DOMException: Failed to set the 'adoptedStyleSheets' property on 'Document': Can't adopt non-constructed stylesheets.` which is to spec so we want to throw as well.
- The `CSSStyleSheet` prototype methods should affect all adopters of that specific stylesheet.